### PR TITLE
Update 60-minute-blitz tutorial

### PR DESCRIPTION
**Pull request by saswatpp**
Flux Tracker functions like back!() and grad() removed in favor of new Zygote's gradient().
Training Loop updated using new Flux.Optimise.update!()

### CODE REVIEW

File: tutorials/60-minute-blitz.jl [hunk 1]
```diff
@@ -119,7 +119,7 @@ f(5)
 # In simple cases it's pretty easy to work out the gradient by hand – here it's
 # `6x+2`. But it's much easier to make Flux do the work for us!
 
-using Flux.Tracker: gradient
+using Flux: gradient
 
 df(x) = gradient(f, x; nest =true)[1]
```
Comment (CarloLucibello) [inline tutorials/60-minute-blitz.jl]: remove `nest`

File: tutorials/60-minute-blitz.jl [hunk 2]
```diff
@@ -171,20 +169,16 @@ gradient(myloss, W, b, x)
...
+W = randn(3, 5)
+b = zeros(3)
 x = rand(5)
 
-y = sum(W * x .+ b)
+y(x) = sum(W * x .+ b)
 
-# Anything marked `param` becomes *tracked*, indicating that Flux keeping an eye
-# on its gradient. We can now call
+grads = gradient(()->y(x), params(W, b))
```
Comment (CarloLucibello) [inline tutorials/60-minute-blitz.jl]:
```suggestion
grads = gradient(()->y(x), Params([W, b]))
```
and add relative import or just `using Flux`

File: tutorials/60-minute-blitz.jl [hunk 3]
```diff
@@ -198,37 +192,34 @@ m = Dense(10, 5)
...
 params(m)
 
-# This makes it very easy to do backpropagation and get the gradient for all
+# This makes it very easy to calculate the gradient for all
 # parameters in a network, even if it has many parameters.
-
+x = rand(Float32, 10)
 m = Chain(Dense(10, 5, relu), Dense(5, 2), softmax)
+l(x) = sum(Flux.crossentropy(m(x), [0.5, 0.5]))
+grads = gradient(Params(params(m))) do
```
Comment (CarloLucibello) [inline tutorials/60-minute-blitz.jl]:
```suggestion
grads = gradient(params(m)) do
```

File: tutorials/60-minute-blitz.jl (hunk 1 above)
Comment (saswatpp) [reply tutorials/60-minute-blitz.jl]: changed in new commit

File: tutorials/60-minute-blitz.jl (hunk 2 above)
Comment (saswatpp) [reply tutorials/60-minute-blitz.jl]: changed in new commit

File: tutorials/60-minute-blitz.jl (hunk 3 above)
Comment (saswatpp) [reply tutorials/60-minute-blitz.jl]: changed in new commit

**Comment by saswatpp** [comment]
The whole Script is Running fine except for one example:
It is related to Zygote Issue: https://github.com/FluxML/Zygote.jl/issues/228
The code is supposed to run fine when the issue is fixed as this is the expected behavior. 
```
mysin(x) = sum((-1)^k*x^(1+2k)/factorial(1+2k) for k in 0:5)

x = 0.5

mysin(x), gradient(mysin, x)
```
Gives error:
```
ERROR: DomainError with -1.0:
log will only return a complex result if called with a complex argument. Try log(Complex(x)).
Stacktrace:
 [1] throw_complex_domainerror(::Symbol, ::Float64) at .\math.jl:31
 [2] log(::Float64) at .\special\log.jl:285
 [3] log at .\special\log.jl:395 [inlined]
 [4] #626 at C:\Users\saswa\.julia\packages\Zygote\tJj2w\src\lib\number.jl:28 [inlined]
 [5] #1659#back at C:\Users\saswa\.julia\packages\ZygoteRules\6nssF\src\adjoint.jl:49 [inlined]
 [6] #3 at .\none:0 [inlined]
 [7] (::typeof(∂(λ)))(::Float64) at C:\Users\saswa\.julia\packages\Zygote\tJj2w\src\compiler\interface2.jl:0
 [8] iterate at .\generator.jl:47 [inlined]
 [9] (::typeof(∂(iterate)))(::Tuple{Float64,Nothing}) at C:\Users\saswa\.julia\packages\Zygote\tJj2w\src\compiler\interface2.jl:0
 [10] mapfoldl_impl at .\reduce.jl:45 [inlined]
 [11] (::typeof(∂(Base.mapfoldl_impl)))(::Float64) at C:\Users\saswa\.julia\packages\Zygote\tJj2w\src\compiler\interface2.jl:0
 [12] mapfoldl_impl at .\reduce.jl:59 [inlined]
 [13] (::typeof(∂(Base.mapfoldl_impl)))(::Float64) at C:\Users\saswa\.julia\packages\Zygote\tJj2w\src\compiler\interface2.jl:0
 [14] mapfoldl at .\reduce.jl:70 [inlined]
 [15] (::typeof(∂(mapfoldl)))(::Float64) at C:\Users\saswa\.julia\packages\Zygote\tJj2w\src\compiler\interface2.jl:0
 [16] #mapreduce#174 at .\reduce.jl:203 [inlined]
 [17] (::typeof(∂(Base.#mapreduce#174)))(::Float64) at C:\Users\saswa\.julia\packages\Zygote\tJj2w\src\compiler\interface2.jl:0
 [18] mapreduce at .\reduce.jl:203 [inlined]
 [19] (::typeof(∂(mapreduce)))(::Float64) at C:\Users\saswa\.julia\packages\Zygote\tJj2w\src\compiler\interface2.jl:0
 [20] sum at .\reduce.jl:397 [inlined]
 [21] mysin at .\REPL[6]:1 [inlined]
 [22] (::typeof(∂(mysin)))(::Float64) at C:\Users\saswa\.julia\packages\Zygote\tJj2w\src\compiler\interface2.jl:0
 [23] gradient at C:\Users\saswa\.julia\packages\Zygote\tJj2w\src\compiler\interface.jl:38 [inlined]
 [24] df(::Float64) at .\REPL[9]:1
 [25] top-level scope at none:0
```

File: tutorials/60-minute-blitz.jl [hunk 4]
```diff
@@ -4,8 +4,8 @@
 # This is a quick intro to [Flux](https://github.com/FluxML/Flux.jl) loosely
 # based on [PyTorch's
 # tutorial](https://pytorch.org/tutorials/beginner/deep_learning_60min_blitz.html).
-# It introduces basic Julia programming, as well as Flux's automatic
-# differentiation (AD), which we'll use to build machine learning models. We'll
+# It introduces basic Julia programming, as well as source-to-source automatic
+# differentiation (AD) in Julia using Zygote Framework, which we'll use to build machine learning models. We'll
 # use this to build a very simple neural network.
```
Comment (CarloLucibello) [inline tutorials/60-minute-blitz.jl]: too much use of "use" in these 3 lines

Comment (saswatpp) [reply tutorials/60-minute-blitz.jl]: changed in new commit

**Comment by CarloLucibello** [comment]
This seems to be the problematic MWE
```julia
julia> f(x) = sum((-1)^k * x^k  for k in 1:1)
f (generic function with 1 method)

julia> gradient(f, 1)
ERROR: DomainError with -1.0:
log will only return a complex result if called with a complex argument. Try log(Complex(x)).
Stacktrace:
 [1] throw_complex_domainerror(::Symbol, ::Float64) at ./math.jl:32
 [2] log(::Float64) at ./special/log.jl:285
 [3] log at ./special/log.jl:395 [inlined]
 [4] #626 at /home/carlo/.julia/packages/Zygote/tJj2w/src/lib/number.jl:28 [inlined]
 [5] #1659#back at /home/carlo/.julia/packages/ZygoteRules/6nssF/src/adjoint.jl:49 [inlined]
 [6] #43 at ./none:0 [inlined]
 [7] (::typeof(∂(λ)))(::Int64) at /home/carlo/.julia/packages/Zygote/tJj2w/src/compiler/interface2.jl:0
 [8] iterate at ./generator.jl:47 [inlined]
 [9] (::typeof(∂(iterate)))(::Tuple{Int64,Nothing}) at /home/carlo/.julia/packages/Zygote/tJj2w/src/compiler/interface2.jl:0
 [10] mapfoldl_impl at ./reduce.jl:55 [inlined]
 [11] (::typeof(∂(mapfoldl_impl)))(::Int64) at /home/carlo/.julia/packages/Zygote/tJj2w/src/compiler/interface2.jl:0
 [12] mapfoldl at ./reduce.jl:72 [inlined]
 [13] (::typeof(∂(mapfoldl)))(::Int64) at /home/carlo/.julia/packages/Zygote/tJj2w/src/compiler/interface2.jl:0
 [14] #mapreduce#194 at ./reduce.jl:200 [inlined]
 [15] (::typeof(∂(#mapreduce#194)))(::Int64) at /home/carlo/.julia/packages/Zygote/tJj2w/src/compiler/interface2.jl:0
 [16] mapreduce at ./reduce.jl:200 [inlined]
 [17] (::typeof(∂(mapreduce)))(::Int64) at /home/carlo/.julia/packages/Zygote/tJj2w/src/compiler/interface2.jl:0
 [18] sum at ./reduce.jl:395 [inlined]
 [19] f at ./REPL[44]:1 [inlined]
 [20] (::typeof(∂(f)))(::Int64) at /home/carlo/.julia/packages/Zygote/tJj2w/src/compiler/interface2.jl:0
 [21] (::Zygote.var"#36#37"{typeof(∂(f))})(::Int64) at /home/carlo/.julia/packages/Zygote/tJj2w/src/compiler/interface.jl:38
 [22] gradient(::Function, ::Int64, ::Vararg{Int64,N} where N) at /home/carlo/.julia/packages/Zygote/tJj2w/src/compiler/interface.jl:47
 [23] top-level scope at REPL[45]:1
```
While these 2 variants work
```julia
julia> f(x) = sum((+1)^k * x^k  for k in 1:1)
f (generic function with 1 method)

julia> gradient(f, 1)
(1,)

julia> f(x) = (-1)^1 * x^1
f (generic function with 1 method)

julia> gradient(f, 1)
(-1,)
```
It seems to be in the family of issue solved by   https://github.com/FluxML/Zygote.jl/pull/512, although it is not clear to me why `(-1)^k` is not treated as a simple coefficient in `sum((-1)^k * x^k  for k in 1:1)`.

Let's see if https://github.com/FluxML/Zygote.jl/pull/512 gets merged soon, we can just show a different example otherwise.

**Comment by saswatpp** [comment]
So should I add a different Example for now ? Or Should we wait for the FluxML/Zygote.jl#512 merge to happen ?

**Comment by CarloLucibello** [comment]
Let's just keep the example how it is,  FluxML/Zygote.jl#512  will be merged soon.
It seems that some functions as `params` and `crossentropy` are used before they are imported, we should fix that

**Comment by saswatpp** [comment]
> It seems that some functions as `params` and `crossentropy` are used before they are imported, we should fix that

`crossentropy` was used as `Flux.crossentropy()`. So I guess `using Flux` would suffice ?

**Comment by saswatpp** [comment]
Training Results:
```
accuracy(valX, valY) = 0.226
accuracy(valX, valY) = 0.327
accuracy(valX, valY) = 0.357
accuracy(valX, valY) = 0.415
accuracy(valX, valY) = 0.433
accuracy(valX, valY) = 0.477
accuracy(valX, valY) = 0.495
accuracy(valX, valY) = 0.509
accuracy(valX, valY) = 0.523
accuracy(valX, valY) = 0.525
10-element Array{Float64,1}:
 0.524
 0.631
 0.261
 0.302
 0.359
 0.525
 0.646
 0.555
 0.675
 0.657
```

**Comment by CarloLucibello** [comment]
thanks!
